File: src/python/compare_images.py
import numpy as np
import math
from scipy.fft import fft2
from PIL import Image

def compare_images(file1, file2):
    image1 = np.array(Image.open(file1).resize((200, 200)).convert('L'))
    image2 = np.array(Image.open(file2).resize((200, 200)).convert('L'))

    fft1 = fft2(image1)
    fft2 = fft2(image2)

    diff = np.mean(np.abs(fft1 - fft2))

    return diff < 0.1  # adjust this threshold as needed

File: src/js/cv/compare-images.js
import sharp from 'sharp';
import { fft2d } from 'fft-js';
import { ndarray, ops } from 'ndarray-fft';
import math from 'mathjs';

export async function compareImages(file1, file2) {
  const image1 = await sharp(file1)
    .resize(200, 200)
    .grayscale()
    .toBuffer();
  
  const image2 = await sharp(file2)
    .resize(200, 200)
    .grayscale()
    .toBuffer();

  const fft1 = fft2d(image1);
  const fft2 = fft2d(image2);
  
  const diff = math.mean(math.abs(math.subtract(ndarray(fft1), ndarray(fft2))));
  
  return diff;
}